spanner/api/Program.cs

```diff
@@ -0,0 +1,989 @@
+            + $"/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd =
+                    connection.CreateSelectCommand("SELECT * FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "
+                        + $"AlbumId : {reader["AlbumId"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END query_data_with_new_column]
+        }
+
+        internal class CustomTransientErrorDetectionStrategy
+            : ITransientErrorDetectionStrategy
+        {
+            public bool IsTransient(Exception ex)
+            {
+                if (ex.IsTransientSpannerFault())
+                    return true;
+                return false;
+            }
+        }
+
+        public static async Task ReadWriteWithTransactionAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START read_write_transaction]
+            // This sample transfers 200,000 from the MarketingBudget 
+            // field of the second Album to the first Album. Make sure to run
+            // the addColumn and writeDataToNewColumn samples first,
+            // in that order.
+
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+
+            using (TransactionScope scope = new TransactionScope(
+                TransactionScopeAsyncFlowOption.Enabled))
+            {
+                decimal transferAmount = 200000;
+                decimal minimumAmountToTransfer = 300000;
+                decimal secondBudget = 0;
+                decimal firstBudget = 0;
+
+                // Create connection to Cloud Spanner.
+                using (var connection =
+                    new SpannerConnection(connectionString))
+                {
+                    // Create statement to select the second album's data.
+                    var cmdLookup = connection.CreateSelectCommand(
+                    "SELECT * FROM Albums WHERE SingerId = 2 AND AlbumId = 2");
+                    // Excecute the select query.
+                    using (var reader = await cmdLookup.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            // Read the second album's budget.
+                            secondBudget =
+                                decimal.Parse(reader["MarketingBudget"]
+                                    .ToString());

```

Comment:
you are missing culture settings -- should use invariant culture.  This is another reason to use GetFieldValue{decimal}("MarketingBudget").  You'll do that in your next PR?